crates/flox/src/commands/package.rs

```diff
@@ -350,22 +381,100 @@ impl PackageCommands {
             },
 
             PackageCommands::Publish2(args) => {
-                let FlakeAttribute {
-                    flakeref,
-                    attr_path,
-                } = args
+                let installable = args
                     .inner
                     .installable_arg
                     .unwrap_or_default()
                     .resolve_flake_attribute(&flox)
                     .await?;
 
-                let publish_ref = PublishFlakeRef::from_flake_ref(flakeref, &flox, false).await?;
-                let publish = Publish::new(&flox, publish_ref, attr_path, config.flox.stability);
-                println!(
-                    "{}",
-                    serde_json::to_string_pretty(publish.analyze().await?.analysis())?
+                let original_flakeref = &installable.flakeref;
+                let publish_flakeref = PublishFlakeRef::from_flake_ref(
+                    installable.flakeref.clone(),
+                    &flox,
+                    args.inner.prefer_https,
+                )
+                .await?;
+
+                if &publish_flakeref != original_flakeref {
+                    info!("Resolved {} to {}", original_flakeref, publish_flakeref);
+                }
+
```

Comment:
I think it could be, the main reason i left it separate was that i initially had `allow_dirty` and `prefer_upstream` flags, which would have pushed `Pulish::new()` to take 7 arguments, 3 of which are not relevant to publishing itself really.
Keeping these steps distinct simplifies the arguments and allows easier error handling IMO